File: backend/components/metiers/administration_publique/templates/core/template_threed.test.js
// template_threed.test.js – Tests unitaires JS pour template métier core
const { render3DTemplateUltra, validate3DTemplateUltra } = require('./template_threed');
describe('TemplateThreed JS (core)', () => {
  test('rendu ultra avancé', () => {
    const data = { id: 1, name: 'Cube', meta: { type: 'test' }, format: 'obj', i18n: { fr: 'Cube', en: 'Cube' } };
    const out = render3DTemplateUltra(data, { audit: 'ok', accessibility: 'ok', rgpd: 'ok' });
    expect(out).toContain('Cube');
    expect(out).toContain('RGPD: ok');
  });
  test('validation ultra avancée', () => {
    expect(validate3DTemplateUltra({ id: 1, name: 'Cube', rgpd: 'ok' })).toBeTruthy();
    expect(validate3DTemplateUltra({ id: 1 })).toBeFalsy();
  });
});

File: backend/components/metiers/administration_publique/templates/core/template_threed.test.py
# template_threed.test.py – Tests unitaires Python pour template métier core
import unittest
from backend.components.metiers.threed.templates.core.template_threed import render_3d_template_ultra, validate_3d_template_ultra

class TestTemplateThreedCore(unittest.TestCase):
    def test_render_ultra(self):
        data = {'id': 1, 'name': 'Cube', 'meta': {'type': 'test'}, 'format': 'obj', 'i18n': {'fr': 'Cube', 'en': 'Cube'}}
        out = render_3d_template_ultra(data, {'audit': 'ok', 'accessibility': 'ok', 'rgpd': 'ok'})
        self.assertIn('Cube', out)
        self.assertIn('RGPD: ok', out)
    def test_validate_ultra(self):
        self.assertTrue(validate_3d_template_ultra({'id': 1, 'name': 'Cube', 'rgpd': 'ok'}))
        self.assertFalse(validate_3d_template_ultra({'id': 1}))

if __name__ == '__main__':
    unittest.main()
